src/truffle-integration/projectDetails.js

```diff
@@ -1,32 +1,43 @@
+const fs = require("fs");
 const path = require("path");
 const child_process = require("child_process");
 const TruffleConfig = require("truffle-config");
+const temp = require("temp");
 
 async function get(projectFile) {
+  temp.track();
+
+  const tempDir = temp.mkdirSync();
```

Comment:
I'm not familiar with the context of this `.get` call or it's timing, so bear with me here. Are the `sync` calls here and in `readFileSync` blocking anything important (like the UI or asynchronous IO in this same process)?

Reply:
You can search the repo for `getProjectDetails` to see the references.

It's used in both the `main` and `truffle-integration` processes, so it does not block any UI (the `renderer` process), and since this function is async and used as an `await` anyway, I would consider it blocking regardless of these calls or not. However, I don't believe it's blocking anything important.